cartographer_ros/cartographer_ros/trajectory_options.cc

```diff
@@ -59,6 +59,12 @@ TrajectoryOptions CreateTrajectoryOptions(
           "num_subdivisions_per_laser_scan");
   options.num_point_clouds =
       lua_parameter_dictionary->GetNonNegativeInt("num_point_clouds");
+  options.rangefinder_sampling_ratio =
+      lua_parameter_dictionary->GetDouble("rangefinder_sampling_ratio");
+  options.odometry_sampling_ratio =
+      lua_parameter_dictionary->GetDouble("odometry_sampling_ratio");
+  options.imu_sampling_ratio =
+      lua_parameter_dictionary->GetDouble("imu_sampling_ratio");
   CheckTrajectoryOptions(options);
```

Comment:
should I check  0 <= ratio <= 1 here, or would you prefer if we introduced a [single check in `FixedRatioSampler`](https://github.com/googlecartographer/cartographer/pull/502/files#diff-eb647468b6eb0fcffce71020e654e168R24)?

Reply:
@wohe no need to check if the ratio is sane?

Reply:
If we add a `CHECK`, we should do it in `FixedRatioSampler()` to avoid the duplication. Doesn't seem like a big issue to me though. Right now we just interpret values clamped to [0, 1].

Reply:
Okay, I have edited the pull request in googlecartographer/cartographer#502, which initially contained the sensor sampling and ratio checking, to retain only the ratio check in `FixedRatioSampler()`.